src/api/schema/components/mod.rs

```diff
@@ -79,6 +87,17 @@ impl filter::CustomFilter<Component> for ComponentsFilter {
     }
 }
 
+impl sort::SortableByField<ComponentsSortFieldName> for Component {
+    fn sort(&self, rhs: &Self, field: &ComponentsSortFieldName) -> cmp::Ordering {
+        match field {
+            ComponentsSortFieldName::Name => Ord::cmp(self.get_name(), rhs.get_name()),
+            ComponentsSortFieldName::ComponentKind => {
+                Ord::cmp(&self.get_component_kind(), &rhs.get_component_kind())
```

Comment:
`Ord + PartialOrd` for enums is definition-order sensitive, which happens to be `Source` -> `Transform` -> `Sink`.

This, conveniently, is sort by `ASC` ordering to return results in what I'd expect (IMO) to be meant by 'component ordering' as a user.

Note that this does _not_ take into account transforms that may have `inputs` pointing from other transforms-- this isn't a 'real' wired up order, per se, just sources appearing before transforms appearing before sinks.